physics/satmedmfvdifq.F

```diff
@@ -1311,6 +1323,9 @@ subroutine satmedmfvdifq_run(ix,im,km,ntrac,ntcw,ntiw,ntke,       &
             do i = 1, im
               qtend = (f2(i,k+is)-q1(i,k,kk))*rdt
               rtg(i,k,kk) = rtg(i,k,kk)+qtend
+              if(kk==ntoz) then
```

Comment:
This change causes the crash for GFSv16beta. Also, all changes in this file will impact the runtime performance - it would be better to have separate loops that are executed only if ldiag3d is true (the if qdiag3d can then stay inside the loops inside the if ldiag3d block, because if someone decides to output 3d tendencies than he/she will expect that the run is slow).

Reply:
I've fixed this, and rearranged the code. I'll test it and let you know when it works.

I had neglected to test with ldiag3d and qdiag3d disabled before.

Reply:
I just merged the latest gsd/develop and I'll test yet again.

Reply:
The deed is done.